packages/angular_devkit/build_angular/src/deploy/index.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { createBuilder } from "@angular-devkit/architect";
+
+const NotImplemented = `
+
+Cannot find "deploy" target for the specified project.
```

Comment:
We could add this message directly on the CLI command instead of creating a dummy builder that is added to all new projects.

Reply:
Without the architect builder the initialize phase of `ArchitectCommand` fails since it cannot find the target in `angular.json`.

Reply:
Sure, but we can catch that error and show the message. It's ok to not have the default targets.

I don't quite agree with that future use of the dummy builder. The approach of using an existing builder to serve as an extension point to other builders subverts the current existing extension point which using builders themselves. With that proposed approach, a separately plugin approach would be needed just for deploy builders so that they could be used by the currently dummy builder.

Reply:
Yeah, I agree with @filipesilva

Reply:
Given the current implementation of the abstract `ArchitectCommand` it doesn't look like it's okay to not have default targets.

When someone invokes `ng deploy --help` the `initialize` method will fail because it'll not find a builder implementation.

Given the current constraints set by `ArchitectCommand.prototype.initialize`, it looks like that's the intended behavior. We can change it, sure.

Overriding the method in `DeployCommand`, catching the exception and handling it is an option but then we should reimplement part of the abstract `initialize`.

Reply:
Sorry I wasn't very clear. What I meant was that it is ok for users to not have (our) default targets. For instance on `--minimal` projects the `test` and `e2e` targets aren't there and that should be handled on our side.

I haven't look at how hard it is to display a custom message for a missing target. But however hard it is I will be easier than introducing new public API (the dummy builder), requiring an update (adding the dummy builder to existing angular.json). So I think we should do it in a way that doesn't require those things, whether through error catching, refactoring the command initialization, or something else. @clydin might have ideas of how to implement this.

Maybe this just means that first party Architect commands that expect the presence of targets need to have better messages in general instead of telling users their angular config is bad.